skimage/morphology/_flood_fill.py

```diff
@@ -0,0 +1,294 @@
+    -------
+    filled : ndarray
+        An array with the same shape as `image` is returned with values equal
+        to (or within tolerance of) the seed point set to `new_value`.
+
+    Notes
+    -----
+    The conceptual analogy of this operation is the 'paint bucket' tool in many
+    raster graphics programs.
+
+    Examples
+    --------
+    >>> from skimage.morphology import flood_fill
+    >>> image = np.zeros((4, 7), dtype=int)
+    >>> image[1:3, 1:3] = 1
+    >>> image[3, 0] = 1
+    >>> image[1:3, 4:6] = 2
+    >>> image[3, 6] = 3
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected ones with 5, with full connectivity (diagonals included):
+
+    >>> flood_fill(image, (1, 1), 5)
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [5, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected ones with 5, with only cardinal direction connectivity:
+
+    >>> flood_fill(image, (1, 1), 5, connectivity=1)
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Fill with a tolerance:
+
+    >>> flood_fill(image, (0, 0), 5, tolerance=1)
+    array([[5, 5, 5, 5, 5, 5, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 5, 5, 3]])
+    """
+    # Correct start point in ravelled image - only copy if non-contiguous
+    image = np.asarray(image)
+    copied = False
+
+    if not image.flags.contiguous:
+        if inplace:
+            warnings.warn('Non-contiguous array passed as `image`; this will '
+                          'be converted to a contiguous array as a copy.')
+            # Flood fill requires some sort of contiguity - this makes a copy
+            image = np.ascontiguousarray(image)
+            copied = True
+
+    if not inplace:
+        if not copied:
+            output = image.copy()
+        else:
+            output = image
+        output[flood_fill_mask(image, seed_point, selem=selem,
+                               connectivity=connectivity,
+                               tolerance=tolerance)] = new_value
+        return output
+    else:  # inplace
+        image[flood_fill_mask(image, seed_point, selem=selem,
+                              connectivity=connectivity,
+                              tolerance=tolerance)] = new_value
+        return image
+
+
+
+def flood_fill_mask(image, seed_point, *, selem=None, connectivity=None,
+                    tolerance=None):
+    """Find mask corresponding to a flood fill applied to an image.
+
+    Starting at a specific `seed_point`, connected points equal or within
+    `tolerance` of the seed value are found, returned as a mask.
+
+    Parameters
+    ----------
+    image : ndarray
+        An n-dimensional array.
+    seed_point : tuple or int
```

Comment:
why "or int"? Is it for the case of 1-D array? I find this "or int" a bit confusing, but maybe I'm missing something.

Reply:
Yes, it's a convenience for the 1-D case.